### PR TITLE
fix: resolved audit report

### PR DESCRIPTION
**Pull request by gummy789j**

### CODE REVIEW

File: contracts/Registry.sol [hunk 1]
```diff
@@ -270,6 +270,13 @@ contract Registry is IRegistry, ERC165 {
         require(modules[_id].contractAddress != address(0), "RRE");
         address origMoudleAddress = modules[_id].contractAddress;
         delete modules[_id];
+        for (uint256 i; i < moduleKeys.length; ++i) {
+            if (moduleKeys[i] == _id) {
+                moduleKeys[i] = moduleKeys[moduleKeys.length - 1];
+                moduleKeys.pop();
+                break;
+            }
+        }
```
Comment (kyle8615) [inline contracts/Registry.sol:281]: use an index mapping to avoid iterating through the whole array?

File: contracts/Timelock.sol [hunk 2]
```diff
@@ -45,21 +45,21 @@ contract Timelock {
...
+    }
 
+    constructor(address _admin, uint256 _delay) ValidDeplay(_delay) {
         admin = _admin;
         delay = _delay;
     }
 
     /// @notice Sets the minimum time delay
     /// @param _delay the new delay
-    function setDelay(uint256 _delay) public onlyAdmin {
```
Comment (kyle8615) [inline contracts/Timelock.sol:58]: admin to gorvernance

**Comment by gummy789j** [comment]
L07 has been solved in commit [fix: MEE](https://github.com/reyield-finance/uniswap-v3-smart-vault/pull/6/commits/98f6a89159a07ee8f08eb4b43dd25d3891a1a590)

**Comment by gummy789j** [comment]
L17 has been solved in commit [fix: L09](https://github.com/reyield-finance/uniswap-v3-smart-vault/pull/6/commits/357c6655b8a8c697192cb0072eed95f0bfee9b75)

**Review by kyle8615 — APPROVED** [review]
LGTM

**Pull request merged**